util/hot_backup.h

```diff
@@ -0,0 +1,127 @@
+// -------------------------------------------------------------------
+//
+// hot_backup.h
+//
+// Copyright (c) 2011-2016 Basho Technologies, Inc. All Rights Reserved.
+//
+// This file is provided to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file
+// except in compliance with the License.  You may obtain
+// a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+// -------------------------------------------------------------------
+
+#ifndef STORAGE_LEVELDB_INCLUDE_HOT_BACKUP_H_
+#define STORAGE_LEVELDB_INCLUDE_HOT_BACKUP_H_
+
+#include "util/thread_tasks.h"
+
+namespace leveldb
+{
+
+namespace config {
+
+// how many hot backup directories:  backup, backup.1, ... backup.(kNumBackups-1)
+static const int kNumBackups = 5;
+
+static const char * kTriggerFileName="/etc/riak/hot_backup";
+
+}   // namespace config
+
+// Called every 60 seconds to test for external hot backup trigger
+//   (initiates backup if trigger seen)
+void CheckHotBackupTrigger();
+
+
+/**
+ * Singleton HotBackup object that unit tests can override.
+ */
+
+extern class HotBackup * gHotBackup;
```

Comment:
Rather than exposing the HotBackup singleton as a global static variable, I recommend making it a private data member of the HotBackup class. Then remove the "unit test support" routines from the public interface of the HotBackup class and add a protected constructor that takes a pointer to the HotBackup object to use as the singleton. For unit tests, have the HotBackupTester class pass its "this" pointer in the base class constructor to set the singleton.

As a general rule, I would like us to move away from having "test only" data members and methods publicly exposed by a class.